scripts/SANS/sans/algorithm_detail/mask_workspace.py

```diff
@@ -127,10 +127,17 @@ def mask_with_mask_files(mask_info, inst_info, workspace):
         load_alg = create_unmanaged_algorithm(load_name, **load_options)
         mask_alg = create_unmanaged_algorithm("MaskDetectors")
 
-        # Masker
-        for mask_file in mask_files:
-            mask_file = find_full_file_path(mask_file)
+        file_paths = [find_full_file_path(i) for i in mask_files]
+        # Find full file path returns an empty string, so we need to remake it
+        missing_file_paths = [mask_files[i] for i, path in enumerate(file_paths) if not path]
+
+        err_str = "The following mask files are missing:"
+        err_str += "\n".join(missing_file_paths)
+        if missing_file_paths:
+            raise FileNotFoundError(err_str)
 
+        # Masker
+        for mask_file in file_paths:
```

Comment:
I may be misunderstanding, but do we not want to exclude the members of `file_paths` that are also in `missing_file_paths` ?

Reply:
In this case we will have already bailed out as the raise winds up the call stack and bails out. We need all masks at this point else faulty tubes will appear in the final reduced data

Reply:
Understood, cheers